quick_select/quick_select.py

```diff
@@ -0,0 +1,61 @@
+def partition(a, start, end):
+    """
+    Perform Partition Operation on array a.
+    Time Complexity: o(nLogn)
+    Auxiliary Space: O(n)
+    :param a: Iterable of elements
+    :param start: pivot value for array
+    :param end: right limit of array
+    :return: return i value for function, used in partitioning of array.
+    """
+    i = start - 1
+    pivot = a[end]
+    for j in range(start, end):
+        if a[j] <= pivot:
+            i += 1
+            a[i], a[j] = a[j], a[i]
+    i += 1
+    a[i], a[end] = a[end], a[i]
+    return i
+
+
+def quick_select(a, k):
+    """
+    Perform quick select operation i.e find kth minimum element from array
+    Time Complexity: O(n) for average case and O(n^2) for worst cases
+    param a: Array on which operation would perfom
+    param k: kth minimun element have to find
+    return: returns kth minimum value
+    """
+    if a is None:
+        print ("Array doesn't exist")
+        return None
+    elif len(a) <= k:
+        print ("k is greater than", len(a) - 1)
+        return None
+    else:
+        start = 0
+        end = len(a)-1
+        is_found = False
+        while not is_found :
+            pos = partition(a, start, end)
+            if pos == k:
+                is_found = True
+                return a[pos]
+            elif pos < k:
+                start = pos + 1
+            else:
+                end = pos - 1
+    return None
+
+
+def main():
+    a = [2, 4, 6, 2, 1, 4, 2, 7, 8, 9, 5, -4, 23, 0, 8]
+    k = 10
+    ans = quick_select(a, k)
+    if ans is not None:
```

Comment:
@Hsankesara Exactly why I'm requesting to add an else block here, and not printing in the function itself. By printing the error in main() you are reducing the flexibility of this code.